components/builder-scheduler/src/data_store.rs

```diff
@@ -391,6 +391,31 @@ impl DataStore {
                                 $$"#,
         )?;
 
+        // Add a target column to the packages table
+        migrator.migrate(
+            "scheduler",
+            r#"ALTER TABLE packages ADD COLUMN IF NOT EXISTS target TEXT DEFAULT NULL"#,
+        )?;
+
+        // Insert or update a new package into the packages table
+        migrator.migrate(
+            "scheduler",
+            r#"CREATE OR REPLACE FUNCTION upsert_package_v1 (
+                                    in_ident text,
+                                    in_deps text[],
+                                    in_target text
+                                ) RETURNS SETOF packages AS $$
+                                        BEGIN
+                                            RETURN QUERY INSERT INTO packages (ident, deps, target)
+                                            VALUES (in_ident, in_deps, in_target)
+                                            ON CONFLICT(ident)
+                                            DO UPDATE SET deps=in_deps, target=in_target RETURNING *;
```

Comment:
Nice!

Reply:
Yeah it's pretty cool.. this capability was added in Postgres 9.5